src/govuk/components/hide-this-page/hide-this-page.yaml

```diff
@@ -0,0 +1,54 @@
+params:
+- name: text
+  type: string
+  required: false
+  description: Text for the link. Defaults to `Hide this page`.
+- name: currentTabURL
+  type: string
+  required: false
+  description: URL to redirect the current tab to. Defaults to `https://www.google.com`.
+- name: newTabURL
+  type: string
+  required: false
+  description: URL to open a new tab to. Defaults to `https://www.bbc.co.uk/weather`.
+- name: fakePageTitle
+  type: string
+  required: false
+  description: Fake title that will be passed to the original page's title attribute to mask the user's history. Defaults to `How to prevent the spread of Coronavirus - GOV.UK`.
+- name: id
+  type: string
+  required: false
+  description: ID attribute to add to the hide this page container.
+- name: classes
+  type: string
+  required: false
+  description: Classes to add to the hide this page container.
+- name: attributes
+  type: object
+  required: false
+  description: HTML attributes (for example data attributes) to add to the hide this page container.
+
+examples:
+- name: default  
+  data:
+    text: Hide this page
+    currentTabURL: "https://www.google.com"
+    newTabURL: "https://www.bbc.co.uk/weather"
+    fakePageTitle: How to prevent the spread of Coronavirus - GOV.UK
```

Comment:
I'd suggest the default  example should only contain any required fields (so in this case, maybe nothing?)

Reply:
I need to work out how best to handle this. Right now if I pass nothing, literally nothing gets applied ie: you get a button with not text in. We should probably set default values for this in the component itself but I've seen different implementations across the repo.

Reply:
Agreed, I'm not sure if I've seen one implementation we always use for this. Potentially it depends on how complex the logic is. I think I've seen some where we do `if params.text else 'default' }}`, but there are instances where we pull out the logic to the top of the template, e.g:

```
{%- if params.text %}
    {% set buttonText = params.text %}
{%- else %}
    {% set buttonText = "default" %}
{%- endif -%}
```

Potentially there's an argument for setting defaults at the top to avoid cluttering the actual HTML code with that logic and hopefully making it easier to read.